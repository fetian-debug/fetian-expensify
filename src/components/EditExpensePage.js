import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import Modal from 'react-modal';

const customStyles = {
  content : {
    top                   : '50%',
    left                  : '50%',
    right                 : 'auto',
    bottom                : 'auto',
    marginRight           : '-50%',
    transform             : 'translate(-50%, -50%)',
    height                : 'auto',
    padding               : '10px 30px 30px 30px',
    borderWidth           : '3px',
    borderRadius          : '10px'
    
  }
};

Modal.setAppElement(document.getElementById('app'));
export class EditExpensePage extends React.Component {
  constructor() {
    super();

    this.state = {
      modalIsOpen: false
    };

    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  openModal() {
    this.setState({modalIsOpen: true});
  }

  closeModal() {
    this.setState({modalIsOpen: false});
  }

  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };

  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
    closeModal();
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
        </div>
        <div className="button__container">
          <button className="button__secondary" onClick={this.openModal}>Remove Expense</button>
        </div>
        <Modal
          isOpen={this.state.modalIsOpen}
          onRequestClose={this.closeModal}
          contentLabel="Remove Expense"
          style={customStyles}
        >
        <div className="modal">
          <h2>Remove the '{this.props.expense.description}' expense?</h2>
          <p>Are you sure you want to remove the {this.props.expense.description} expense?</p>
          <p>Warning: This cannot be undone!</p>
          <div className="button-modal__container">
            <button className="button__success" onClick={this.closeModal && this.onRemove}>Yes</button>
            <button className="button__fail" onClick={this.closeModal}>No</button>
          </div>
        </div>
        </Modal>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
