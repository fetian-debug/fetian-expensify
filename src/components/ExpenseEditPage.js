import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {startEditExpense, startRemoveExpense} from '../actions/expenses';


export class ExpenseEditPage extends React.Component {

    onClick = () => { 
        this.props.startRemoveExpense({ id: this.props.Expense.id } );
        this.props.history.push('/');

    };
 
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.Expense.id, expense);
        this.props.history.push('/');
    };

    render() {
        return (
                <div>
                 <div className="page-header">
                  <div className="content-container">
                   <h1 className="page-header__title">Edit Expense</h1>                         
                  </div>
                 </div>
                 <div className="content-container">
                  <ExpenseForm
                   expense={this.props.Expense}
                   onSubmit={ this.onSubmit }
                  />
                  <button className="button button--secondary" onClick={ this.onClick }>
                  Remove Expense
                  </button>
                 </div>
                 
                </div>

          );
    };

};

const mapStateToProps = (state, props) => { 
    return {Expense: state.expenses.find( (expense) => props.match.params.id === expense.id )};
};

const mapDispatchToProps = (dispatch, props) => {
    return {
        startEditExpense: (id,expense) => { dispatch(startEditExpense(id,expense)) },
        startRemoveExpense: (data) => {dispatch(startRemoveExpense(data))}
    };
};
    



export default connect(mapStateToProps,mapDispatchToProps)(ExpenseEditPage);