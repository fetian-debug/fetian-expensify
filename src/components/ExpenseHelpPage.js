import React from 'react';

const ExpenseHelpPage = () => ( 
    <div>
     This is from my expense help page
    </div>
);



export default ExpenseHelpPage;