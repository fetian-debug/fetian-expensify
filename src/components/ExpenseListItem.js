import React from 'react';
import {Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral'; 

const ExpenseListItem = (props) => ( // props can be destructed to be {description, amount, createdAt, id}
    
     <Link className="list-item" to={`/edit/${props.id}`}>
      <div>
       <h3 className="list-item__title">{props.description}</h3>
       <span className="list-item__subtitle">{moment(props.createdAt).format('MMMM Do, YYYY')}</span> 
      </div>
      <div>
       <h3 className="list-item__data">{numeral(props.amount /100).format('$0,0.00')}</h3>
      </div>
     </Link>
);

    // const mapStateToProps = (state) => {


    //     return {
    //         description: state.expenses.description,
    //         amount: state.expenses.amount ,
    //         createdAt: state.expenses.createdAt 
            
    //     };
    // };

    
 
export default ExpenseListItem; //only to access props.dispatch();


