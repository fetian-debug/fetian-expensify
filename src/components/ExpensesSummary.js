import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal, allExpensesTotal, expenses }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses' ;
  let formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  const expenseFullWord = expenses - expenseCount === 1 ? 'expense' : 'expenses';
  const formattedHiddenExpenses = numeral((allExpensesTotal - expensesTotal) / 100).format('$0,0.00');
  const formattedAllExpenses = numeral(allExpensesTotal / 100).format('$0,0.00');
  const hiddenExpenseCount = expenses.length - expenseCount;
  
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
        {expensesTotal !== allExpensesTotal && 
          <h3 className="page-header__subtitle">There {hiddenExpenseCount === 1 ? 'is ' : 'are '} 
          <span>{hiddenExpenseCount}</span> hidden {hiddenExpenseCount === 1 ? 'expense ' : 'expenses '} 
          totalling <span>{formattedHiddenExpenses}</span></h3>}
        <div className="page-header__actions">
          <Link className="button" to="/create">Add Expense</Link>
        </div>
      </div>
    </div>
  );
};

const resetFilter = () => {
  expensesTotal = allExpensesTotal;
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses),
    allExpensesTotal: selectExpensesTotal(state.expenses),
    expenses: state.expenses
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
