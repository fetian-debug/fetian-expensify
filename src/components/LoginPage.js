import React from 'react';
import { connect } from 'react-redux';
import { startLoginGoogle, startLoginFacebook, startLoginGitHub, startLoginTwitter } from '../actions/auth';

export const LoginPage = ({ startLoginGoogle, startLoginFacebook, startLoginGitHub, startLoginTwitter }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>It's time to get your expenses under control</p>
      <button onClick={startLoginGoogle} className="button__login button__login--google"><i className="fa fa-google" aria-hidden="true"></i>Login With Google</button>
      <button onClick={startLoginGitHub} className="button__login button__login--github"><i className="fa fa-github" aria-hidden="true"></i>Login With Github</button>
      <button onClick={startLoginFacebook} className="button__login button__login--facebook"><i className="fa fa-facebook" aria-hidden="true"></i>Login With Facebook</button>
      <button onClick={startLoginTwitter} className="button__login button__login--twitter"><i className="fa fa-twitter" aria-hidden="true"></i>Login With Twitter</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLoginGoogle: () => dispatch(startLoginGoogle()),
  startLoginFacebook: () => dispatch(startLoginFacebook()),
  startLoginGitHub: () => dispatch(startLoginGitHub()),
  startLoginTwitter: () => dispatch(startLoginTwitter())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
