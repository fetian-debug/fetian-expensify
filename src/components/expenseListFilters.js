import React from 'react';
import {connect} from 'react-redux';
import {setupTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates'


export class ExpenseListFilters extends React.Component {
    
    state = {
        focusedInput: null
    };

    onDatesChange = ({startDate,endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);

    };

    onFocusChange = (calendarFocused) =>{
        this.setState(()=>({calendarFocused }))
    };

    onTextChange = (e) => {
        this.props.setupTextFilter(e.target.value);
    };

    onSelectChange = (e) => {
        (e.target.value === 'date' ? this.props.sortByDate() : this.props.sortByAmount() );
        
    };
   
    render () {
       return  (
            <div className="content-container">
             <div className="input-group">
              <div className="input-group__item">
               <input className='text-input' type="text" placeholder="Search Expense" value={this.props.filters.text} onChange={this.onTextChange}/>
              </div>
              <div className="input-group__item">
               <select className='select' value={this.props.filters.sortBy} onChange={this.onSelectChange}>    
                <option  disabled>sort</option>
                <option value="date">Date</option>
                <option value="amount">Amount</option>
               </select>
              </div>
              <div className="input-group__item">
               <DateRangePicker
                startDate={this.props.filters.startDate}
                endDate={this.props.filters.endDate}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.focusedInput}
                onFocusChange={this.onFocusChange}
                numberOfMonths={1}
                isOutsideRange={()=>false}
                showClearDates={true}
               />
              </div>
             
             </div>
            </div>
        );
             }
};





const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
}; 

const mapDispatchToProps = (dispatch) => {
    return {
        setStartDate: (startDate) => { dispatch(setStartDate(startDate)) },
        setEndDate: (endDate) => {dispatch(setEndDate(endDate))},
        setupTextFilter: (value) => {dispatch(setupTextFilter(value))},
        sortByDate: () => {dispatch(sortByDate())},
        sortByAmount: () => {dispatch(sortByAmount())}

    };
};


export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);