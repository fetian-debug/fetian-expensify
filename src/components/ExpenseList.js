import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import getVisibleExpenses from '../selectors/expenses';
import {Link} from 'react-router-dom';



export const ExpenseList = (props) => (
    <div className="content-container">

     <div className="list-header">
      <div className="show-for-mobile">Expenses</div>
      <div className="show-for-desktop">Expesne</div>
      <div className="show-for-desktop">Amount</div>
      
     </div>
      <div className="list-body">
      
    {
        props.expenses.length === 0 ? (
            <div className="list-item list-item--message"> 
             <span>No items found</span>
            </div>
            
        ) : (
            props.expenses.map((expense) =>{ 
        return  <ExpenseListItem key={expense.id} {...expense}/> ;
    })
        )
    }
    </div>
   </div>
);

// key={expense.id} {...other}
//          expenseDesc={expense.description}
//          expenseAmount={expense.amount}
//          expenseCreatedAt={expense.createdAt}
//          expenseId= {expense.id}

const mapStateToProps = state => ({
    expenses: getVisibleExpenses(state.expenses, state.filters) 
});

export default connect(mapStateToProps)(ExpenseList); //here we get something back: not the hoc but the function  

