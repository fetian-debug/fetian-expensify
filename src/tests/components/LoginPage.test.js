import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should correctly render LoginPage', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
  const startLoginGoogle = jest.fn();
  const wrapper = shallow(<LoginPage startLoginGoogle={startLoginGoogle} />);
  wrapper.find('.button__login--google').simulate('click');
  expect(startLoginGoogle).toHaveBeenCalled();
});
