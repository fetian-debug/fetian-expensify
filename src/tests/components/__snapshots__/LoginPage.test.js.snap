// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LoginPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      It's time to get your expenses under control
    </p>
    <button
      className="button__login button__login--google"
    >
      <i
        aria-hidden="true"
        className="fa fa-google"
      />
      Login With Google
    </button>
    <button
      className="button__login button__login--github"
    >
      <i
        aria-hidden="true"
        className="fa fa-github"
      />
      Login With Github
    </button>
    <button
      className="button__login button__login--facebook"
    >
      <i
        aria-hidden="true"
        className="fa fa-facebook"
      />
      Login With Facebook
    </button>
    <button
      className="button__login button__login--twitter"
    >
      <i
        aria-hidden="true"
        className="fa fa-twitter"
      />
      Login With Twitter
    </button>
  </div>
</div>
`;
